### PR TITLE
Add ability to specify row key

### PR DESCRIPTION
**Pull request by joellanciaux**

### CODE REVIEW

File: src/utils/dataUtils.js [hunk 1]
```diff
@@ -22,28 +22,28 @@ function fromJSGreedy(js) {
...
-  // build up a new list of data and list lookup
-  for (let i = 0; i < data.length; i += 1) {
-    const key = getKey();
-    lookup[key] = i;
-
-    // get an Immutable map of the data item
-    const map = fromJSGreedy(data[i]);
-    list.push(localSettings.addGriddleKey ? map.set(localSettings.name, key) : map);
-  }
+export function transformData(data, renderProperties) {
```
Comment (ryanlanciaux) [inline src/utils/dataUtils.js]: Does this make `addKeyToCollection` no longer necessary?

File: src/reducers/dataReducer.js [hunk 2]
```diff
@@ -49,7 +48,7 @@ export function GRIDDLE_INITIALIZED(initialState) {
  * @param {Object} action - The action object to work with
 */
 export function GRIDDLE_LOADED_DATA(state, action) {
-  const transformedData = transformData(action.data);
+  const transformedData = transformData(action.data, state.get('renderProperties').toJSON());
```
Comment (ryanlanciaux) [inline src/reducers/dataReducer.js]: (I personally like the idea of throwing an error saying something like `given rowKey does not exist` so it's really clear why it's failing)

Comment (joellanciaux) [inline src/reducers/dataReducer.js]: We _can_, but I think it'd going to lead to something that's a bit of a bummer during `GRIDDLE_INITIALIZE`, where the initial conversion into Immutable happens.

The alternative to `toJSON()` here would be converting the `renderProperties` in `GRIDDLE_INITIALIZE` to a `Map` before the rest of the initial state and then executing `set('renderProperties', renderPropertiesMap)`.

Both options seem pretty lame, heh.

File: src/utils/dataUtils.js (hunk 1 above)
Comment (joellanciaux) [inline src/utils/dataUtils.js]: yep -- looks like you yanked it out.

Comment (ryanlanciaux) [inline src/utils/dataUtils.js]: :sunglasses:

File: src/reducers/dataReducer.js (hunk 2 above)
Comment (ryanlanciaux) [inline src/reducers/dataReducer.js]: Got it -- I think this is good then.

**Pull request merged**